evcache-core/src/main/java/net/spy/memcached/EVCacheMemcachedClient.java

```diff
@@ -104,7 +106,21 @@ public EVCacheMemcachedClient(ConnectionFactory cf, List<InetSocketAddress> addr
                 .orElseGet("evcache.get.alwaysDecodeSync")
                 .orElse(true);
         this.alwaysDecodeSync = alwaysDecodeSyncProperty.get();
-        alwaysDecodeSyncProperty.subscribe(v -> alwaysDecodeSync = v);
+
+        // Subscription is maintained so that we can release the subscription on shutdown,
+        // otherwise it will retain a reference to this instance and prevent garbage collection.
+        this.alwaysDecodeSyncSubscription = alwaysDecodeSyncProperty.subscribe(v -> alwaysDecodeSync = v);
+    }
+
+    @Override
+    public boolean shutdown(long timeout, TimeUnit unit) {
+        try {
+            return super.shutdown(timeout, unit);
+        } finally {
+            if (alwaysDecodeSyncSubscription != null) {
+                alwaysDecodeSyncSubscription.unsubscribe();
```

Comment:
Thanks @jasonk000 for the fix.

For history: This was the [commit](https://github.com/Netflix/EVCache/commit/60e2a773198e75e2605eeda61a0a9944386ec316) that introduced this behavior. I overlooked the unsubscribe part during the code review as the other classes that do subscribe are long living objects (like EVCacheClientPool etc) and don't do explicit unsubscribe(). 

These are short living objects especially when we have flakey connections which would end up leaking and cause OOM issues.